infra: uncomment code blocks programmatically for testing, remove randomness notebook from excluded list

virajvchaudhari opened the pull request:
…domness notebook from excluded list

*Issue #, if available:*

*Description of changes:*
- uncomments the notebook commented cell blocks based on the identifier tag so that can run the test end to end for the notebook. 
- currently removed one notebook from the excluded list. 
- ran test with and without least mocking options. 


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

rmshaffer commented on test/integ_tests/test_all_notebooks.py:
```diff
@@ -178,6 +178,54 @@ def execute_with_mocks(tb, mock_level, path_to_utils, path_to_mocks):
+    for line in lines:
+        stripped_line = line.strip()
+
+        if stripped_line == "# UNCOMMENT_FOR_TEST":
```
I don't see where you have this string in the notebook. Should it be the same as the string above, `"# UNCOMMENT_TO_RUN"`? (If so, maybe define that string at the file scope, to use in both places?)

virajvchaudhari replied:
Yes, I am good to change that to descriptive comment.

virajvchaudhari replied:
@rmshaffer Oh wait, I missed out changing the tag here. Let me fix that.

yitchen-tim commented:
This is awesome! I added a section in Testing.md to document the behavior.

virajvchaudhari commented on examples/advanced_circuits_algorithms/Randomness/Randomness_Generation.ipynb:
```diff
@@ -710,7 +710,7 @@
   },
```
<pre class="ql-syntax" spellcheck="false">Line #26.    # UNCOMMENT_TO_RUN
</pre><p><span class="mention" data-value="rmshaffer" data-denotation-char="@">﻿<span contenteditable="false">@rmshaffer</span>﻿</span> the comment tag has been added here. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/amazon-braket/amazon-braket-examples/pull/705/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

virajvchaudhari commented:
> This is awesome! I added a section in Testing.md to document the behavior.



> This is awesome! I added a section in Testing.md to document the behavior.

Thank you @yitchen-tim for doing that!

rmshaffer reviewed (approved):
🥳

rmshaffer commented on TESTING.md:
```diff
@@ -75,3 +75,16 @@ Simply append the following line to the code cell specified in the Recording sec
+
+In the example notebooks, some code cells are presented as commented out to prevent accidental execution of costly or long-running operations, particularly those that run on a QPU. However, commented code can become prone to errors over time due to lack of regular testing.
+
+To improve test coverage, we introduce a special flag: `# UNCOMMENT_TO_RUN`. As a developer, you can include this flag within a cell to mark the beginning of a block that should be uncommented during integration testing. The integration test framework will detect the # UNCOMMENT_TO_RUN marker and automatically uncomment all subsequent lines until it encounters a line that is not commented.
```
```suggestion
To improve test coverage, we introduce a special flag: `## UNCOMMENT_TO_RUN`. As a developer, you can include this flag within a cell to mark the beginning of a block that should be uncommented during integration testing. The integration test framework will detect the `## UNCOMMENT_TO_RUN` marker and automatically uncomment all subsequent lines until it encounters a line that is not commented.
```
